### PR TITLE
Refactor Local Storage interaction with database

### LINKED ISSUE #198: Refactor Frontend Start
Currently at modeler start there will be a request sent to the backend to retrieve all available applications, and then get the available tasks for each application, as that one is selected.
Additionally, there is a request to get all <application|task> tuples at the very beginning, which is needed for cross-checking the parameters of activities.

Each is stored separately and there are multiple requests going back an forth.

---

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] Should remove the calls to get all applications and the tasks and render those elements needed for the session storage based on all the <application|task> tuples

### PR DESCRIPTION
**Pull request by lukashueller**
## Changes
- changes all localStorage Names to `camelCase`
- remove **"getAvailableApplicaions()"** Backend-Call for retrieving `availableApplications` and replace it with an iteration over `taskAndApplicationCombinations`
- combine `robotName` and `robotId` into a `robotMetadata` object => set `robotName` when clicking on a RobotContainer-Card
- remove caching and the session storage entry `taskApplicationCache` => retrieve all information from `taskAndApplicationCombinations`

closes #198

### CODE REVIEW

File: frontend/src/utils/sessionStorage/localSsotController/ssot.js [hunk 1]
```diff
@@ -23,22 +23,34 @@ import {
...
  * @returns Currently saved robotId
  */
-const getRobotId = () => JSON.parse(sessionStorage.getItem(ROBOT_ID_PATH));
+const getRobotId = () =>
+  JSON.parse(sessionStorage.getItem(ROBOT_METADATA_PATH)).robotId;
 
 /**
- * @description Writes the robotId of the currently opened robot into the session storage
+ * @description Gets the robotName of the currently opened robot from the session storage
+ * @returns Currently saved robotName
```
Comment (SanJSp) [inline frontend/src/utils/sessionStorage/localSsotController/ssot.js:39]: Can you add a type here? I guess `{String}` is the type, isn't it?

File: frontend/src/utils/sessionStorage/localSsotController/ssot.js [hunk 2]
```diff
@@ -23,22 +23,34 @@ import {
...
  * @description Gets the robotId of the currently opened robot from the session storage
  * @returns Currently saved robotId
  */
-const getRobotId = () => JSON.parse(sessionStorage.getItem(ROBOT_ID_PATH));
+const getRobotId = () =>
+  JSON.parse(sessionStorage.getItem(ROBOT_METADATA_PATH)).robotId;
 
 /**
- * @description Writes the robotId of the currently opened robot into the session storage
+ * @description Gets the robotName of the currently opened robot from the session storage
```
Comment (SanJSp) [inline frontend/src/utils/sessionStorage/localSsotController/ssot.js:38]:
```suggestion
 * @description Retrieves the robotName of the currently opened robot from the session storage
```

**Review by SanJSp — APPROVED** [review]
Two minor documentation hints, other than that good work :)

File: frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js [hunk 3]
```diff
@@ -27,9 +29,9 @@ import { parseBpmnToSsot } from '../../../../../utils/parser/bpmnToSsotParsing/b
  * @param {Object} modeler the modeling object
  * @param {String} robotId id of the robot
```
Comment (WolfgangDaniel) [inline frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js:30]: This line should be deleted

File: frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js [hunk 4]
```diff
@@ -64,31 +66,24 @@ const modelerElementChangeHandler = (event, elementState, setterObject) => {
 };
 
 /**
- * @description Checks if tasks for selected application are already stored in session storage.
- * Otherwise, fetch tasklist from MongoDB.
+ * @description Sets all tasks for currently selected application from session storage
  * @param {String} application Application for which to get the tasks for.
- * @param {Object} setterObject object containing the functions for setting the state in the React component
+ * @param {Object[]} taskApplicationCombinations Array of task and application combination objects.
```
Comment (WolfgangDaniel) [inline frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js:71]: Shouldn't the type be Array?

**Review by WolfgangDaniel — CHANGES_REQUESTED** [review]
Just some minor changes. Apart from that it looks fantastic. Thanks for the great job 👍

File: frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js (hunk 3 above)
Comment (WolfgangDaniel) [reply frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js:30]: line 30

File: frontend/src/utils/sessionStorage/localSsotController/ssot.js (hunk 1 above)
Comment (lukashueller) [reply frontend/src/utils/sessionStorage/localSsotController/ssot.js:39]: fixed

File: frontend/src/utils/sessionStorage/localSsotController/ssot.js (hunk 2 above)
Comment (lukashueller) [reply frontend/src/utils/sessionStorage/localSsotController/ssot.js:38]: fixed

File: frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js (hunk 3 above)
Comment (lukashueller) [reply frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js:30]: you are right! i've just deleted it!

File: frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js (hunk 4 above)
Comment (lukashueller) [reply frontend/src/components/pages/RobotModeler/ModelerSidebar/modelerSidebarFunctionality/modelerSidebarFunctionality.js:71]: Also fixed. i wasn't shure how to write "Array of objects" but I think just a simple `Array` is right.

**Review by WolfgangDaniel — APPROVED** [review]

**Pull request merged**